feat - User(Auth)

Hodu-C opened the pull request:
gradle - security 의존성 추가 SecurityConfig - BCrypt 암호화 사용 - 패스워드 암호화 테스트를 위해 cors, csrf, formlogin, headers 비활성화 - 회원가입, 로그인 서비스 User -> Auth 도메인 변경 UserRepo - 이메일 중복 방지 존재 여부 확인 추가 AuthService - 회원가입 수정(패스워드 암호화)

Hodu-C committed:
feat - User(Auth)

gradle
- security 의존성 추가
SecurityConfig
- BCrypt 암호화 사용
- 패스워드 암호화 테스트를 위해 cors, csrf, formlogin, headers 비활성화
- 회원가입, 로그인 서비스 User -> Auth 도메인 변경
UserRepo
- 이메일 중복 방지 존재 여부 확인 추가
AuthService
- 회원가입 수정(패스워드 암호화)

Heo-y-y commented on src/main/java/com/melog/melog/domain/auth/dto/AuthSignUpDto.java:
```diff
@@ -7,7 +7,7 @@
 
 
-public record UserSignUpDto(
+public record AuthSignUpDto(
```
record와 class를 통일 시키는 것이 좋아 보입니다.

Hodu-C replied:
수정하였습니다.

Heo-y-y commented on src/main/java/com/melog/melog/domain/auth/service/AuthService.java:
```diff
@@ -1,12 +1,73 @@
+
+    private void validateSignUpDto(AuthSignUpDto authSignUpDto) {
+        if (userRepository.existsByEmail(authSignUpDto.email())) {
+            throw new BusinessException(ErrorCode.EMAIL_ALREADY_EXISTS);
```
ErrorCode 상위로 임포트하면 좋을 것 같아요

Hodu-C replied:
수정하였습니다.

Heo-y-y commented on src/main/java/com/melog/melog/domain/auth/service/AuthService.java:
```diff
@@ -1,12 +1,73 @@
+        userRepository.save(user);
+    }
+
+    private void validateSignUpDto(AuthSignUpDto authSignUpDto) {
```
해당 메서드가 여러 곳에서 쓰일까요? 그게 아니라면 signUp 메서드에서 한번에 처리하는 것이 좋아보입니다.

Heo-y-y commented on src/main/java/com/melog/melog/domain/user/controller/UserController.java:
```diff
@@ -25,14 +23,8 @@ public ResponseEntity<Object> info(@PathVariable Long id){
     @PatchMapping("/update/{id}")
-    public ResponseEntity<Object> update(@PathVariable Long id, @Valid @RequestBody UserUpdateDto userUpdateDto) throws Exception {
+    public ResponseEntity<Object> update(@PathVariable Long id, @Valid @RequestBody UserUpdateDto userUpdateDto){
         User data = userService.update(id, userUpdateDto);
```
엔티티를 바로 보내는 것은 보안상 좋지 않아보입니다.
그리고 변수 이름도 data 보다는 user가 좋아보여요

Hodu-C replied:
user 도메인 수정 시 수정하겠습니다.

Heo-y-y commented on src/main/java/com/melog/melog/domain/user/service/UserService.java:
```diff
@@ -1,16 +1,15 @@
 import com.melog.melog.domain.user.dto.request.UserUpdateDto;
 import com.melog.melog.domain.user.entity.User;
 
 public interface UserService {
```
각 코드별 impl과 class 방식을 통일 시키는 방법이 좋아보입니다.

Hodu-C replied:
class 로 통일하였습니다.

Heo-y-y commented on src/main/java/com/melog/melog/domain/user/service/UserServiceImpl.java:
```diff
@@ -18,25 +17,16 @@ public class UserServiceImpl implements UserService{
-    public User update(Long id, UserUpdateDto userUpdateDto) throws Exception {
+    public User update(Long id, UserUpdateDto userUpdateDto){
         User user = userRepository.findById(id).orElseThrow(()->
                 new BusinessException(ErrorCode.USER_NOT_FOUND));
```
에러코드 import *

Hodu-C replied:
수정하였습니다.

Heo-y-y commented on src/main/java/com/melog/melog/global/error/ErrorCode.java:
```diff
@@ -6,7 +6,7 @@
 @Getter
 @AllArgsConstructor
 public enum ErrorCode {
```
게시물 용, 인증 용, 회원 용 주석으로 나눠주는 것이 보기 편할 것 같습니다.

Hodu-C replied:
수정하였습니다!

Heo-y-y commented:
변경 부분 푸쉬하셨나요~?

Hodu-C committed:
Refactor - User(Auth)

DTO - class와 record 였던 것들 record 로 통합
반복되는 메시지 처리 import static 으로 코드 간소화
인터페이스 -> 구현체 구조, 인터페이스 제거 통일
에러코드 - 주석 추가

Hodu-C commented:
푸쉬했습니다. 다음 푸쉬는 시간이 조금 걸릴 거 같습니다 ㅠ
